components/function-controller/pkg/controller/function/function_controller.go

```diff
@@ -106,385 +98,436 @@ var (
-	_                 reconcile.Reconciler = &ReconcileFunction{}
-
-	// "build and push step"
-	buildAndPushStep = "build-step-build-and-push"
+	// name of build Task
+	buildTaskName = getEnvDefault("BUILD_TASK", "function-build")
 )
 
-// ReconcileFunction is the controller.Reconciler implementation for Function objects
-// ReconcileFunction reconciles a Function object
+// ReconcileFunction is the controller.Reconciler implementation for Function objects.
 type ReconcileFunction struct {
 	client.Client
 	scheme *runtime.Scheme
 }
 
-func getEnvDefault(envName string, defaultValue string) string {
+func getEnvDefault(envName, defaultValue string) string {
 	// use default value if environment variable is empty
-	var value string
-	if value = os.Getenv(envName); value == "" {
-		return defaultValue
+	if value := os.Getenv(envName); value != "" {
+		return value
 	}
-	return value
+	return defaultValue
 }
 
 // Reconcile reads that state of the cluster for a Function object and makes changes based on the state read
 // and what is in the Function.Spec
 // +kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=serviceaccounts,verbs=get;watch;update;list
+// +kubebuilder:rbac:groups="apps;extensions",resources=deployments,verbs=create;get;watch;update;delete;list;update;patch
 // +kubebuilder:rbac:groups="serverless.kyma-project.io",resources=functions,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="serverless.kyma-project.io",resources=functions/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups="admissionregistration.k8s.io",resources=mutatingwebhookconfigurations;validatingwebhookconfigurations,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="serving.knative.dev",resources=services;routes;configurations;revisions,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="build.knative.dev",resources=builds;buildtemplates;clusterbuildtemplates;services,verbs=get;list;create;update;delete;patch;watch
-// +kubebuilder:rbac:groups="",resources=serviceaccounts,verbs=get;watch;update;list
-// +kubebuilder:rbac:groups=";apps;extensions",resources=deployments,verbs=create;get;watch;update;delete;list;update;patch
-func (r *ReconcileFunction) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+// +kubebuilder:rbac:groups="tekton.dev",resources=tasks;taskruns,verbs=get;list;watch;create;update;patch;delete
+func (r *ReconcileFunction) Reconcile(req reconcile.Request) (reconcile.Result, error) {
 	// Get Function instance
-	fn := &serverlessv1alpha1.Function{}
-	err := r.getFunctionInstance(request, fn)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return reconcile.Result{}, nil
+	fn, err := r.getFunctionInstance(req)
+	switch {
+	case errors.IsNotFound(err):
```

Comment:
doesn't isNotFound() also mean that the function is probably not even created and then you flag it for creation?

Reply:
The first thing we do in Reconcile() is checking whether the function exists, we would have returned early if the Function didn't exist.